components/system/components/GlobalCarousel.js

```diff
@@ -155,20 +162,16 @@ export class GlobalCarousel extends React.Component {
   };
 
   _handleOpen = (e) => {
-    let carouselType =
-      !this.props.current ||
-      (this.props.current &&
-        (this.props.current.decorator === "FOLDER" || this.props.current.decorator === "ACTIVITY"))
-        ? "data"
-        : "slate";
+    if (e.detail.index < 0 || e.detail.index >= this.props.objects.length) {
```

Comment:
This condition, albeit useful, is a little difficult to read, can we improve it?